src/main/java/net/schmizz/sshj/connection/channel/direct/SessionChannel.java

```diff
@@ -93,7 +98,7 @@ public Command exec(String command)
             throws ConnectionException, TransportException {
         checkReuse();
         log.debug("Will request to exec `{}`", command);
-        sendChannelRequest("exec", true, new Buffer.PlainBuffer().putString(command))
+        sendChannelRequest("exec", true, new Buffer.PlainBuffer().putString(command.getBytes(getRemoteCharset())))
```

Comment:
The main question I have when reading this, should _every_ string be written using the remote charset? If so, it is nicer to make it a property of the Buffer, rather than converting it everywhere we call `putString`.